src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -285,16 +266,127 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBody
                 context.Write(writer, new DynamicJsonValue
                 {
                     [nameof(BatchCommandResult.Results)] = array,
-                    [nameof(BatchCommandResult.TransactionIndex)] = result.Index
+                    [nameof(BatchCommandResult.TransactionIndex)] = index
                 });
             }
         }
 
+        private async Task<long?> GetClusterTransactionCount(object result, string raftRequestId, long databaseCommandsCount, Task<long?> onDatabaseCompletionTask)
+        {
+            if (databaseCommandsCount == 0)
+                return null;
+
+            long? count = null;
+
+            if (result != null && result is ClusterTransactionResult clusterTxResult)
+            {
+                if(clusterTxResult.Count.HasValue == false)
+                    throw new InvalidOperationException($"Cluster Transaction result is null, but has more then 0 database commands.");
+
+                count = clusterTxResult.Count.Value;
+            
+                Database.ForTestingPurposes?.AfterCommitInClusterTransaction?.Invoke();
+            
+                using (var token = CreateHttpRequestBoundOperationToken())
+                    await onDatabaseCompletionTask.WithCancellation(token.Token);
```

Comment:
we always wait for this, no? 
let's move it outside then

Reply:
also, if it returns value we are good.. not matter the case, right?

Reply:
We can't because in the first 'if' we need to do it in the end of the scope, and in the 'else' you need to do it in the start of the scope.

Reply:
I _think_ you can always wait for this (when `databaseCommandsCount > 0`) ... please check it.

Reply:
Done